swarm/api/http/server.go

```diff
@@ -290,6 +291,111 @@ func (s *Server) HandleDelete(w http.ResponseWriter, r *Request) {
 	fmt.Fprint(w, newKey)
 }
 
+func (s *Server) HandlePostDb(w http.ResponseWriter, r *Request) {
+	if r.ContentLength == 0 {
+		frequency, err := strconv.ParseUint(r.uri.Path, 10, 64)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			http.ServeContent(w, &r.Request, "", time.Now(), bytes.NewReader([]byte(err.Error())))
+			return
+		}
+		err = s.api.DbCreate(r.uri.Addr, frequency)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	} else {
+		data, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		_, _, _, err = s.api.DbUpdate(r.uri.Addr, data)
+		if err != nil {
+			w.WriteHeader(http.StatusUnauthorized)
```

Comment:
Because if it fails it is assumed that the caller doesn't have access to edit the resource (false signature, for example). 

Do we need this more granular? Like storage timeout error, for example?

Reply:
Is that the _only_ error that can happen though? I think this kind of assumption needs to be commented, it is not very obvious.

Reply:
Yeah, it's not. And it's not the only error that's currently returned. We need error types here.